radicle-node/src/service/session.rs

```diff
@@ -363,4 +368,16 @@ impl Session {
         }
         Ok(())
     }
+
+    pub(crate) fn add_pending_fetch(&mut self, rid: Id) {
+        self.pending_fetches.push(rid);
+    }
+
+    pub(crate) fn next_pending_fetch(&mut self) -> Option<Id> {
+        if self.pending_fetches.is_empty() {
+            None
+        } else {
+            Some(self.pending_fetches.remove(0))
+        }
```

Comment:
It would make more sense to use a `VecDeque` with `pop_front()` and `push_back()`